src/main/java/com/nickdsantos/onedrive4j/OneDrive.java

```diff
@@ -168,15 +168,21 @@ public AccessToken getAccessTokenFromRefreshToken(String refreshToken) throws IO
 			HttpPost httpPost = new HttpPost(uri);
 			httpPost.setEntity(formEntity);
 
-			Map<Object, Object> rawToken = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
-			if (rawToken != null) {
+			Map<Object, Object> rawResponse = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
+
+			if (rawResponse.containsKey("error")) {
```

Comment:
`rawResponse` could be null here and you'll get an NPE here if that happens. If it's not null here, it means that it will not be null when it gets to line 178.